File: apps/desktop/src/renderer/src/api/libraries.ts
import { apiGet, apiPost } from "./client";

export interface Library {
  id: number;
  name: string;
  root_path: string;
  created_at: string;
  is_active: boolean;
}

export const libraries = {
  init: (path: string) => apiPost<Library>("/api/libraries/init", { path }),
  active: () =>
    apiGet<Library>("/api/libraries/active").catch((e) => {
      if (String(e).includes("404")) return null;
      throw e;
    }),
  list: () => apiGet<Library[]>("/api/libraries"),
};

File: packages/beatos-http/beatos_http/routes/libraries.py
"""/api/libraries routes.

Three endpoints — no /activate by id. Switching the active library = POSTing
/init with the target path. Identification is by root_path globally.
"""
from __future__ import annotations

from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

from beatos_core.library.service import (
    get_active_library,
    init_library_root,
    list_libraries,
)
from beatos_core.models import Library

router = APIRouter(prefix="/api/libraries", tags=["libraries"])


class InitPayload(BaseModel):
    path: str


@router.post("/init", response_model=Library)
async def init(payload: InitPayload) -> Library:
    return await init_library_root(payload.path)


@router.get("/active", response_model=Library)
async def active() -> Library:
    lib = await get_active_library()
    if lib is None:
        raise HTTPException(status_code=404, detail="No active library.")
    return lib


@router.get("", response_model=list[Library])
async def list_all() -> list[Library]:
    return await list_libraries()
